flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/batch/connectors/cassandra/CassandraColumnarOutputFormatBase.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.batch.connectors.cassandra;
+
+import org.apache.flink.streaming.connectors.cassandra.ClusterBuilder;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.guava30.com.google.common.base.Strings;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.io.IOException;
+
+/**
+ * CassandraColumnarOutputFormatBase is the common abstract class for writing into Apache Cassandra
+ * using column based output formats.
+ *
+ * @param <OUT> Type of the elements to write.
+ */
+public abstract class CassandraColumnarOutputFormatBase<OUT>
+        extends CassandraOutputFormatBase<OUT, ResultSet> {
+    private final String insertQuery;
+    private transient PreparedStatement prepared;
+
+    public CassandraColumnarOutputFormatBase(String insertQuery, ClusterBuilder builder) {
+        super(builder);
+        Preconditions.checkArgument(
+                !Strings.isNullOrEmpty(insertQuery), "Query cannot be null or empty");
+        this.insertQuery = insertQuery;
+    }
+
+    /**
+     * Opens a Session to Cassandra and initializes the prepared statement.
+     *
+     * @param taskNumber The number of the parallel instance.
+     * @throws IOException Thrown, if the output could not be opened due to an I/O problem.
```

Comment:
seems strange to document the parameters when this is already done in the interface.